File: test-suite/src/test/java/io/micronaut/kafka/docs/producer/config/ProductClient.java
package io.micronaut.kafka.docs.producer.config;

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.context.annotation.Property;
import io.micronaut.context.annotation.Requires;
import org.apache.kafka.clients.producer.ProducerConfig;
// end::imports[]

@Requires(property = "spec.name", value = "ProductClientTest")
// tag::clazz[]
@KafkaClient(
    id = "product-client",
    acks = KafkaClient.Acknowledge.ALL,
    properties = @Property(name = ProducerConfig.RETRIES_CONFIG, value = "5")
)
public interface ProductClient {
    // define client API
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/kafka/docs/producer/config/ProductClient.py
from abc import ABC

# tag::imports[]
from micronaut.configuration.kafka.annotation import KafkaClient
from micronaut.context.annotation import Property, Requires
from org.apache.kafka.clients.producer import ProducerConfig
# end::imports[]


@Requires(property="spec.name", value="ProductClientTest")
# tag::clazz[]
@KafkaClient(
    id="product-client",
    acks=KafkaClient.Acknowledge.ALL,
    properties=Property(name=ProducerConfig.RETRIES_CONFIG, value="5")
)
class ProductClient(ABC):
    # define client API
    pass
# end::clazz[]
